### PR TITLE
Milestone 2: Setup

### PR DESCRIPTION
**Pull request by Lizdev-05**
In this milestone, I achieved the following...

- Set up associations between models.
- Used models to insert data
- Created custom models
- Users
    - A method that returns the 3 most recent posts for a given user.
- Posts
    - A method that updates the posts counter for a user.
    - A method that returns the 5 most recent comments for a given post.
 - Comments
    - A method that updates the comments counter for a post.
 - Likes
    - A method that updates the likes counter for a post.

### CODE REVIEW

**Comment by julie-ify** [comment]
Hi @Lizdev-05,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### Kindly fix your linter error

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Lizdev-05** [comment]
Thank you @julie-ify , I have fixed the linters

File: app/models/like.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class Like < ApplicationRecord
+  belongs_to :post
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
```
Comment (t-yanick) [inline app/models/like.rb:3]: - [ ] As it is a [requirement ](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project_v1_1.md)for this project, " `Likes:
A method that updates the likes counter for a post.` ", kindly add a method that updates the likes counter for a post. Something like so: 

```ruby
def update_likes_counter
  post.increment!(:likes_counter)
end
```

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @Lizdev-05, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/like.rb (hunk 1 above)
Comment (Lizdev-05) [reply app/models/like.rb:3]: Ok, I will work on it. Thank you

**Review by m05e5 — APPROVED** [review]
Hi @Lizdev-05 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
goodjob

## Highlights
✔️ your linters are all ok
✔️ Descriptive PR 

## Optional suggestions


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Lizdev-05** [comment]
Thank you for reviewing and approving my work

**Pull request merged**